observer/mon_conf.go

```diff
@@ -0,0 +1,63 @@
+package observer
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/letsencrypt/boulder/cmd"
+	p "github.com/letsencrypt/boulder/observer/probes"
+	"gopkg.in/yaml.v2"
+)
+
+type settings map[string]interface{}
+
+// MonConf is exported to receive yaml configuration
+type MonConf struct {
+	Valid    bool
+	Period   cmd.ConfigDuration `yaml:"period"`
+	Timeout  int                `yaml:"timeout"`
+	Probe    string             `yaml:"type"`
+	Settings settings           `yaml:"settings"`
+}
+
+func (c MonConf) normalize() {
+	c.Probe = strings.ToLower(c.Probe)
+}
+
+func (c MonConf) unmashalProbeSettings() (p.Configurer, error) {
+	probeConf, err := p.GetProbeConf(c.Probe, c.Settings)
+	if err != nil {
+		return nil, err
+	}
+	s, _ := yaml.Marshal(c.Settings)
```

Comment:
Feels weird to convert already-parsed yaml to a string, so that we can parse it again. I'm honestly not sure what the best alternative here is, but it seems worth investigating.

Reply:
Marshaling it to bytes so that I can unmarshal it into the correct ProberConf (DNS/HTTP). I found it to be a good alternative to writing a map to struct for DNS and HTTP that calls reflect.Type() for each field.